### PR TITLE
Use Gitlab Container registry

### PR DESCRIPTION
**Pull request by mkuron**
This pull request builds our images on Gitlab (https://gitlab.icp.uni-stuttgart.de/espressomd/docker/container_registry). This means our build machines no longer need to talk to Docker Hub and can instead get the images over our fast internal network. Furthermore, we can now build images that we may not build on Docker Hub -- such as one with the Intel Compiler and Intel MPI, as included in this pull request.

### CODE REVIEW

**Comment by mkuron** [comment]
Sure, I can do that, but then you need to set up all the CI webhook stuff for this repo too.

**Comment by KaiSzuttor** [comment]
I just have to push PRs too, right?

**Comment by mkuron** [comment]
Yes, but we also need the status reporting scripts that post the pending/success/failure status to Github.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  tags:
+    - docker
+    - privileged
+
+.test_template: &test_definition
+  stage: test
+  before_script:
+    - docker pull $CI_REGISTRY/$CI_PROJECT_PATH/$CI_JOB_NAME:latest
+  script:
+    - git clone https://github.com/espressomd/espresso
```
Comment (KaiSzuttor) [inline .gitlab-ci.yml]: this has to happen within the container as before

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    - docker
+    - privileged
+
+.test_template: &test_definition
+  stage: test
+  image: $CI_REGISTRY/$CI_PROJECT_PATH/$CI_JOB_NAME:$CI_COMMIT_SHA
+  script:
+    - git clone https://github.com/espressomd/espresso
+    - cd espresso
+    - maintainer/travis/build_cmake.sh
```
Comment (KaiSzuttor) [inline .gitlab-ci.yml]: maintainer/CI/build_cmake.sh

**Comment by mkuron** [comment]
Ready to merge. The last failing test will turn green once https://github.com/espressomd/espresso/pull/1912 is merged. Also, please disable Travis-CI for this repository.

**Comment by hmenke** [comment]
Travis should be disabled now.

**Pull request merged**